source/config.go

```diff
@@ -35,6 +42,24 @@ type Config struct {
 }
 
 // Validate executes manual validations beyond what is defined in struct tags.
-func (c Config) Validate() error {
-	return c.Config.Validate()
+func (c *Config) Validate(ctx context.Context) error {
+	var multierr []error
+	err := c.Config.Validate()
+	if err != nil {
+		multierr = append(multierr, err)
+	}
+	// validate and set the topics.
+	if len(c.Topic) == 0 && len(c.Topics) == 0 {
+		multierr = append(multierr, fmt.Errorf("required parameter missing: %q", "topics"))
+	}
+	if len(c.Topic) > 0 && len(c.Topics) > 0 {
+		multierr = append(multierr, fmt.Errorf(`can't provide both "topic" and "topics" params, "topic" will be deprecated soon, please only provide the "topics" parameter instead`))
+	}
+	if len(c.Topic) > 0 && len(c.Topics) == 0 {
+		sdk.Logger(ctx).Warn().Msg(`"topic" parameter will be deprecated soon, please use "topics" instead.`)
```

Comment:
On the same note, the field is already deprecated.
```suggestion
		sdk.Logger(ctx).Warn().Msg(`"topic" parameter is deprecated and will be removed, please use "topics" instead.`)
```

Reply:
Oh, and same in the error above in line 56.